src/traces/parcoords/attributes.js

```diff
@@ -0,0 +1,188 @@
+                'of `[fromValue, toValue]` with finite numbers as elements.'
+            ].join(' ')
+        },
+        values: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'Dimension values. `values[n]` represents the value of the `n`th point in the dataset,',
+                'therefore the `values` vector for all dimensions must be the same (longer vectors',
+                'will be truncated). Each value must be a finite number.'
+            ].join(' ')
+        },
+        description: 'The dimensions (variables) of the parallel coordinates chart. 2..63 dimensions are supported.'
+    },
+
+    tickdistance: {
+        valType: 'number',
+        dflt: 50,
+        min: 32,
+        role: 'style',
+        description: 'The desired approximate tick distance (in pixels) between axis ticks on an axis.'
+    },
+
+    padding: {
+        valType: 'number',
+        dflt: 80,
+        min: 0,
+        role: 'style',
+        description: 'The desired space for displaying axis labels and domain ranges around the actual parcoords.'
+    },
+
+    line: extendFlat({},
+
+        colorAttributes('line'),
+
+        {
+            showscale: {
+                valType: 'boolean',
+                role: 'info',
+                dflt: false,
+                description: [
+                    'Has an effect only if `marker.color` is set to a numerical array.',
+                    'Determines whether or not a colorbar is displayed.'
+                ].join(' ')
+            },
+            colorbar: colorbarAttrs
+        },
+
+        {
+            contextopacity: {
+                valType: 'number',
+                dflt: 0.05,
+                min: 0,
+                max: 1,
+                role: 'style',
+                description: 'Opacity of the context lines, on a scale of 0 (invisible) to 1 (fully opaque).'
+            },
+
+            focusopacity: {
+                valType: 'number',
+                dflt: 1,
+                min: 0,
+                max: 1,
+                role: 'style',
+                description: 'Opacity of the focus lines, on a scale of 0 (invisible) to 1 (fully opaque).'
+            },
+
+            pixelratio: {
+                valType: 'number',
+                dflt: 1,
+                min: 0.25,
+                max: 4,
+                role: 'style',
+                description: 'Line rendering pixel ratio. A lower value yields faster rendering but blockier lines.'
+            },
+
+            focusalphablending: {
+                valType: 'boolean',
+                dflt: false,
+                role: 'style',
+                description: [
+                    'By default, the rendered lines are opaque. Setting it to `true` is necessary if opacity is needed.'
+                ].join(' ')
+            }
+        }
+    ),
+
+    blocklinecount: {
```

Comment:
This one too, no need to expose this.

For now, you should put these values in a `src/traces/parcoords/constants.js`, If a power users really wants to tweak those values, we'll be able to point that users to the correct file and ask them build their own plotly.js bundle.